src/_pytest/main.py

```diff
@@ -423,6 +430,10 @@ def pytest_ignore_collect(collection_path: Path, config: Config) -> bool | None:
         if any(fnmatch_ex(pat, collection_path) for pat in norecursepatterns):
             return True
 
+        if any(fnmatch_ex(pat, collection_path) for pat in ("build", "dist")):
```

Comment:
should this be wrapped by a "allow_in_setuptools" like they do for a venv?

Reply:
i have rewritten this in light of what I understood from our conversation, please let me know if this is the direction we want to go in